namespace Throwing
{
    using System;

    class MyArgumentException : Exception { }


    public class NumberGenerator
    {
        int[] numbers = {2, 4, 6, 8, 10, 12, 14, 16, 18, 20};

        public int GetNumber(string indexString)
        {
            try
            {
                var index = int.Parse(indexString);

                if (index == 2)
                {
                    throw new MyArgumentException();
                }

                try
                {
                    return numbers[index];
                }
                catch (IndexOutOfRangeException)
                {
                    throw new NullReferenceException();
                }
            }
            catch (Exception e)
            {
                throw new MyArgumentException();
            }
        }
    }
}
